### DISCUSSION TITLE
PrismaClient is not configured

### DISCUSSION THREAD

**Discussion by FaseehHayat21** [prisma/prisma#25427] [Q&A]
### Bug description

Error: PrismaClient is not configured to run in Edge Runtime (Vercel Edge Functions, Vercel Edge Middleware, Next.js (Pages Router) Edge API Rou
tes, Next.js (App Router) Edge Route Handlers or Next.js Middleware). In order to run Prisma Client on edge runtime, either:
- Use Prisma Accelerate: https://pris.ly/d/accelerate
- Use Driver Adapters: https://pris.ly/d/driver-adapters

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 

### Prisma Version

```

```

**Comment by ludralph** [comment]
Hi @FaseehHayat21 👋 

Next.js middlewares always run in the **Edge runtime**, not in a traditional Node.js environment. As a result, you'll need to use Prisma's Edge Function support and won't be able to rely on the classic built-in drivers, even for local development. You can find more information in the Next.js documentation: [Next.js Middleware Runtime](https://nextjs.org/docs/app/building-your-application/routing/middleware#runtime).

To run Prisma on Edge runtimes (such as Vercel Edge Functions), you'll need to enable the `driverAdapters` preview feature. This involves using Prisma's **Driver Adapter** and passing an instance of it to `@prisma/client`. 

For more details, please check out Prisma's [blog post](https://www.prisma.io/blog/prisma-orm-support-for-edge-functions-is-now-in-preview) and the official [documentation](https://www.prisma.io/docs/orm/prisma-client/deployment/edge/deploy-to-vercel).

Hope this helps!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
